PAS-46 Adds prettier and eslint to Passr

polygonart6777 opened the pull request:
-Adds eslint and prettier to the configuration file
-Adds npm commands to lint the project, one with fixing one with not
-Spcifies configuration for VSCode to auto lint but not to auto save
lint changes and includes recommended extensions.
-Adds pre-commit hook to re-format files that are marked as "staged"
before you commit.

polygonart6777 committed:
PAS-46 (WIP) Adds prettier and eslint to Passr
-Adds eslint and prettier to the configuration file
-Adds npm commands to lint the project, one with fixing one with not
-Spcifies configuration for VSCode to auto lint but not to auto save
lint changes and includes recommended extensions.
-Adds pre-commit hook to re-format files that are marked as "staged"
before you commit.

polygonart6777 committed:
PAS-46 WIP deletes duplicate line and added simpler formatting to regex

BraidenCutforth commented:
@polygonart6777 you should rebase this work on master to pull in my github action. It runs the `npm run lint` command as part of the action

BraidenCutforth commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+	"editor.defaultFormatter": "esbenp.prettier-vscode",
+	"editor.formatOnSave": true,
+	"eslint.validate": ["javascript", "typescript", "typescriptreact"]
+}
```
I am always unsure if this is something we should impose this way? Granted I think there is a HUGE benefit in having this occur by default. I just wonder if someone goes to change their vscode config, if it would be annoying to have to always deal with trying not to commit it

BraidenCutforth commented on eslintrc.js:
```diff
@@ -0,0 +1,35 @@
+		'react/no-unused-prop-types': 'warn',
+		'react-hooks/rules-of-hooks': 'error',
+		'react-hooks/exhaustive-deps': 'warn',
+		'prettier/prettier': 'warn',
```
Not sure if this rule means that prettier linting will only emit a warning? We probably want it to error out.

BraidenCutforth commented on package.json:
```diff
@@ -1,40 +1,73 @@
+		"test": "react-scripts test",
+		"eject": "react-scripts eject",
+		"eslint": "eslint 'src/**/*.{t,j}s{x,}'",
+		"eslint:fix": "eslint 'src/**/*.{t,j}s{x,}', --fix'",
```
I know we are using eslint to do the linting, but I believe we should keep the actual commands that we run as `lint` and `lint:fix` as the idea behind the action is not dependent on the tool we use to do it

BraidenCutforth commented on package.json:
```diff
@@ -1,40 +1,73 @@
+	},
+	"eslintConfig": {
+		"extends": "react-app"
+	},
```
I think we should use a separate [prettier configuration] file, like `.prettierrc.js`because I don't want to over bloat our package.json. It will make prettier changes easier to find in the future too. 

Same applied for the eslintConfig. I think you can add this to the .eslintrc.js

BraidenCutforth commented on package.json:
```diff
@@ -1,40 +1,73 @@
+			"eslint",
+			"prettier --write"
+		]
+	}
```
Does husky have an option for its own configuration file? Same reason as the prettier config

BraidenCutforth commented on src/App.tsx:
```diff
@@ -3,24 +3,24 @@ import logo from './logo.svg';
-      </header>
-    </div>
-  );
+	return (
```
I've always been a 4 space/tab kinda guy like what is done here, I am just wondering, does this seem excessively spaced? Have I been wrong all along? It always looks different in an editor than GitHub though so, I am not sure

BraidenCutforth reviewed (commented):
Looks great as a WIP, just saw it and I jumped the gun a bit but was just too excited for this ticket that I had to start talking 🤣

polygonart6777 committed:
PAS-46 Adds documentation and a test for lint-staging

polygonart6777 committed:
PAS-46 testing git hook with eslint

polygonart6777 committed:
PAS-46 (WIP) Adds better lint configuration file.  Documentation stil needs to be finished with justification for each configuration tool.

polygonart6777 committed:
PAS-46 (WIP) changes name for eslint to lint

polygonart6777 committed:
PAS-46 updated react-scripts and dependencies in order to help build run sucessfully

polygonart6777 committed:
PAS-46 updated react-scripts and dependencies in order to help build run sucessfully

polygonart6777 committed:
Merge branch 'PAS-46-add-eslint-prettier' of https://github.com/VikeLabs/Passr into PAS-46-add-eslint-prettier

polygonart6777 committed:
PAS-46 finished documentation

polygonart6777 committed:
PAS-46 removes eslint git commit check and adjusts the documentation

BraidenCutforth commented on .gitignore:
```diff
@@ -17,7 +17,10 @@ node_modules
 .env.test.local
 .env.production.local
+.eslintcache
+.vscode
```
So, I am not sure if this is an attempt to stop future changes, or stop the changes you have added here in the branch, but since `.vscode/extensions.json` and `.vscode/settings.json` have been checked in already, I think any changes made in the future will still be permitted. Ie, the gitignore has no effect on them.

BraidenCutforth commented on .eslintignore:
```diff
@@ -0,0 +1,3 @@
+*.test.tsx
+*.ts
```
Shouldn't we lint these files as well?

BraidenCutforth commented on .eslintignore:
```diff
@@ -0,0 +1,3 @@
+*.test.tsx
+*.ts
+.eslintrc.js
```
Also not a _big_ deal, but you can set your editor to trim extra new lines at the end of files and also always make sure there is a new line

polygonart6777 commented:
@BraidenCutforth in reference to comment about the .eslintignore file above, right now ESLint throw a `no-undef` error for the `test` and `expect` functions.  We can either disable this rule for all files, ignore test files or ignore the issue in each specific file.  For now, I thought this was the best solution.  Let me know if you have any other suggestions for this issue.

polygonart6777 committed:
PAS-46 Addresses code review comments
-Removes unnecessary files from gitignore
-Addresses typos in documentation

polygonart6777 committed:
PAS-46 Addresses code review comments
-Installs testing library plugin so that we can lint test files
-Addresses eslint errors by moving functions before they are used
-Updates plugins to the most current version

polygonart6777 committed:
PAS-46 Fixes typos in documentation and removes unnecessary files from gitignore.

polygonart6777 committed:
PAS-46 Adds new npm scripts to the ReadMe file

polygonart6777 committed:
PAS-46- Removes double word from ReadMe. Why must I spot typos right after I push up changes?

polygonart6777 committed:
PAS-46 Adds path fix for Windows
